### PR TITLE
feat: init sns

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

- init Sns library
- refactor candid to handle did files in a single folder for the all mono-repo

# Changes

- init new package `packages/sns` (information copied from https://github.com/dfinity/sns-js)
- refactor existing candid files to root `./candid/`
- add `didc` typescript generation to `./scripts/compile-idl-js` 
- clean up candid files to apply same convention across canisters and use `didc` to generate typescript file

a. `something.did` -> the candid definition
b. `something.d.ts` -> the typescript definition for the types and service of the did files (generated with `didc`)
c. `something.idl.js` -> the factory js file (generated with `didc`)
d. `something.idl.d.ts` -> the typescript definition of the above factory file
e. `something.certified.idl.js` -> the factory js file for the certified calls (generated with `didc`)
f. `something.certified.idl.d.ts` -> the typescript definition of the above factory file

note: all these above files and content are now generated automatically by `./scripts/compile-idl-js` 

- update typescript imports according above naming and structure
- add candid files for Sns (root, sns governance and swap. ledger is the same as nns)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,8 +7,8 @@ The libraries are still in active development, and new features will incremental
 ## Libraries
 
 - [nns](/packages/nns): interfacing with the **ledger** and **governance** canisters of the IC
-- sns-js: interacting with a Sns project (_coming soon_)
-- cmc-js: querying the Cmc canister (_coming soon_)
+- [sns](/packages/sns): interacting with a sns project (_coming soon_)
+- cmc: querying the Cmc canister (_coming soon_)
 
 ## Integration with NNS-dapp
```
Comment (lmuntaner) [inline README.md:14]: Can you add here or in ahoter readme the comments you did about the candid files as well:

```
a. something.did -> the candid definition
b. something.d.ts -> the typescript definition for the types and service of the did files (generated with didc)
c. something.idl.js -> the factory js file (generated with didc)
d. something.idl.d.ts -> the typescript definition of the above factory file
e. something.certified.idl.js -> the factory js file for the certified calls (generated with didc)
f. something.certified.idl.d.ts -> the typescript definition of the above factory file
```

That is very helpful.

File: packages/nns/src/index.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-export type { DeployedSns } from "../candid/sns_wasm.idl";
+export type { DeployedSns } from "../../../candid/sns_wasm";
```
Comment (lmuntaner) [inline packages/nns/src/index.ts:1]: is the sns_wasm the SNS Manager mentioned in other schemas?

**Review by lmuntaner — APPROVED** [review]
LGTM, thanks!

Just two minor questions.

File: packages/nns/src/index.ts (hunk 2 above)
Comment (peterpeterparker) [reply packages/nns/src/index.ts:1]: yes

it changed names a couple of times but is now (according the answer I got about it last week from nns team) named "sns-wasm"

File: README.md (hunk 1 above)
Comment (peterpeterparker) [reply README.md:14]: sure done, I included, extended and refactored the `HOW-To.md` to `HACKING.md`.

i gonna merge the PR to make/test these changes available for nightly-build but if you notice anything related that should be improved, let me know. happy to provide another PR.

**Pull request merged**